aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java

```diff
@@ -91,37 +93,26 @@ public void initialize(Map<String, String> properties) {
     Preconditions.checkNotNull(
         region, "Cannot initialize AssumeRoleClientConfigFactory with null region");
 
-    this.s3Endpoint = properties.get(AwsProperties.S3FILEIO_ENDPOINT);
     this.tags = toTags(properties);
-    this.s3UseArnRegionEnabled =
-        PropertyUtil.propertyAsBoolean(
-            properties,
-            AwsProperties.S3_USE_ARN_REGION_ENABLED,
-            AwsProperties.S3_USE_ARN_REGION_ENABLED_DEFAULT);
-    this.dynamoDbEndpoint = properties.get(AwsProperties.DYNAMODB_ENDPOINT);
-    this.httpClientType =
-        PropertyUtil.propertyAsString(
-            properties, AwsProperties.HTTP_CLIENT_TYPE, AwsProperties.HTTP_CLIENT_TYPE_DEFAULT);
-  }
-
-  protected <T extends AwsClientBuilder & AwsSyncClientBuilder> T configure(T clientBuilder) {
-    AssumeRoleRequest request =
+    this.assumeRoleRequest =
         AssumeRoleRequest.builder()
             .roleArn(roleArn)
             .roleSessionName(genSessionName())
```

Comment:
I am not sure about this part, could you check if we need a different session name for each assume role request?

Reply:
I did some search and here is what I found:

The `roleSessionName` is used to differentiate between different programs which assume the same role. It seems that a single program can make requests with the same `roleSessionName` to establish different types of connections. Hence, I think we can use the same session name here.

It seems that we can instead treat this change as a new feature and add some functionality such as optional user-defined `roleSessionName`. Hence, I plan to leave this unchanged in this PR and do it in another.

The doc related to `roleSessionName`: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/STS.html#assumeRole-property

Reply:
yes I agree that session name should be configurable in 2 ways:
1. the session name itself
2. a boolean flag indicating if session name should be random, which is done by appending a random UUID suffix

However, I think the session names in the request used by the `StsAssumeRoleCredentialsProvider ` should be consistent and not changing for each request. Considering from the audit log perspective, we should expect the same session name to show even if there are multiple assume role requests sent to refresh the credential, because they all come from the same program from start to end. So I think `roleSessionName` should be an instance variable with a fixed name.

Reply:
Thank you for your explanation. In this case, I think we can make the whole request an instance variable since all the configurations are fixed after the initialization step.